### DISCUSSION TITLE
Not assuming proper role on EKS

### DISCUSSION THREAD

**Discussion by hariharan614** [aws/aws-sdk-php-laravel#244] [Q&A]
Hi All, We are running the laravel application on EKS, and attached the service account to it

 I was trying below things, but couldn't get it working as expected 

1. When running standalone php script to check the role it's assuming, it was picking service account
2. But when tried via the application, it was picking the node instance role

I'm using the default provider with out any credentials defined.

Any help would be appreciated

**Comment by trinm1912** [comment]
I have the same issue.
Does anyone have any idea about it?
cc: @jeremeamia @howardlopez

**Comment by Robert-Tran-Geocomply** [comment]
I also faced this issue when deploying service on EKS. I'm really appreciated with any support

**Comment by Robert-Tran-Geocomply** [comment]
@hariharan614 I found a solution that worked for my case and hope it can help you.

When you configure a `serviceAccountName` in your deployment, Kubernetes will define some environment variables like
```
      AWS_STS_REGIONAL_ENDPOINTS:           regional
      AWS_DEFAULT_REGION:                   us-east-1
      AWS_REGION:                           us-east-1
      AWS_ROLE_ARN:                         arn:aws:iam::<account-id>:role/<role-name>
      AWS_WEB_IDENTITY_TOKEN_FILE:          /var/run/secrets/eks.amazonaws.com/serviceaccount/token
``` 

You should add these vars to your .env file
